src/script/calculator.ts

```diff
@@ -0,0 +1,488 @@
+
+
+  constructor(screenId: string) {
+    this.screen = document.getElementById(screenId) as HTMLElement;
+    if (!this.screen) {
+      throw new Error(`Screen element with id "${screenId}" not found.`);
+    }
+    this.calculationDone = false;
+    this.screen.textContent = localStorage.getItem("calculationOutput") || "0";
+    this.isDegreeMode = true;
+    this.updateDegButton();
+    this.isPrimary = false;
+    this.FEMode = false;
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+    this.math = Math as ExtendedMath;
+    
+
+    // Constants for buttons
+    const FEButton = document.getElementById("fe-btn") as HTMLElement;
+    const sinBtn = document.querySelector(".sin-btn") as HTMLElement;
+    const cosBtn = document.querySelector(".cos-btn") as HTMLElement;
+    const tanBtn = document.querySelector(".tan-btn") as HTMLElement;
+
+    if (!FEButton) {
+      console.error("F-E Button not found!");
+    }
+
+    this.FEButton = FEButton;
+    this.sinBtn = sinBtn;
+    this.cosBtn = cosBtn;
+    this.tanBtn = tanBtn;
+  }
+
+  public handleKeyPress(this: Calculator, event: KeyboardEvent): void {
+    const key: string = event.key;
+
+    if (!isNaN(Number(key))) {
+      this.appendValue(key);
+    } else if (key === "+") {
+      this.add();
+    } else if (key === "-") {
+      this.subtract();
+    } else if (key === "*") {
+      this.multiply();
+    } else if (key === "/") {
+      this.divide();
+    } else if (key === "Enter") {
+      this.result();
+    } else if (key === "Backspace") {
+      this.backspace();
+    } else if (key === "Escape" || key.toLowerCase() === "c") {
+      this.clearDisplay();
+    }
+  }
+
+  appendValue(value: string): void {
+    const currentText = this.screen.textContent || "";
+    const operators = ["+", "-", "×", "÷", ".", "!", "%"];
+    const lastChar = currentText.slice(-1);
+
+    // Check if the input exceeds the 20 character limit
+    if (currentText.length >= 20) {
+      alert("Cannot exceed more than 20 input values");
+      return;
+    }
+
+    if (this.calculationDone) {
+      this.screen.textContent = "";
+      this.calculationDone = false;
+    }
+
+    if (operators.includes(lastChar) && operators.includes(value)) {
+      return;
+    }
+
+    if (
+      value === "." &&
+      (lastChar === "." ||
+        currentText
+          .split(/[\+\-\*\%\!/]/)
+          .pop()
+          ?.includes("."))
+    ) {
+      alert("Cannot enter multiple decimal values");
+      return;
+    }
+
+    if (this.screen.textContent === "0" && !operators.includes(value)) {
+      this.screen.textContent = value;
+    } else {
+      this.screen.textContent += value;
+    }
+    this.screen.scrollTo(this.screen.offsetWidth, 0);
+  }
+
+  initializeMemoryFunctions(): void {
+    // Constants for memory function buttons
+    const mcBtn = document.querySelector(".mc-btn") as HTMLElement;
+    const mrBtn = document.querySelector(".mr-btn") as HTMLElement;
```

Comment:
Applicable for all remaining buttons.

```suggestion
    const mrBtn = document.querySelector(".mr-btn") as HTMLButtonElement;
```

Reply:
add https://github.com/Sakshi-Simform/calculator-ts/commit/b3685c79ca00f5bfb8e5bfae751b6118cd6a7522